prow/git/git.go

```diff
@@ -175,7 +184,7 @@ func (c *Client) Clone(repo string) (*Repo, error) {
 		Dir:    t,
 		logger: c.logger,
 		git:    c.git,
-		base:   base,
+		base:   gitURL.String(),
```

Comment:
This won't reload the secrets if they get changed, do we care?

Reply:
I'm not sure. The gitURL will always be up to date, right? It always calls `c.getCredentials` whenever a `Clone` is made as L143